internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go

```diff
@@ -89,15 +89,20 @@ func resourceMongoDBAtlasFederatedSettingsIdentityProviderRead(ctx context.Conte
 
 	ids := conversion.DecodeStateID(d.Id())
 	federationSettingsID := ids["federation_settings_id"]
-	oktaIdpID := ids["okta_idp_id"]
+
+	// Since the migration of this resource to the latest version of the auto-generated SDK (v20231115) and the breaking changes of the API
+	// the unique identifier used by the API & SDK of the supported identity providers is no longer "okta_idp_id", it is "idp_id". Nonetheless
+	// "okta_idp_id" name was used to encode/decode the Terraform State Id. To avoid increasing the complexity of the code and be able to make
```

Comment:
```suggestion
	// "okta_idp_id" name was used to encode/decode the Terraform State Id. To ensure backwards compatibility, the format of this resource id remains the same but the key `okta_idp_id` will store either `okta_idp_id` or `idp_id` to identify the identity provider.
```
small suggestion to be more precise on the motivation of leaving this inconsistency

Reply:
Done